sonar-db/src/main/java/org/sonar/db/user/UserDao.java

```diff
@@ -177,11 +198,27 @@ public UserDto selectOrFailByLogin(DbSession session, String login) {
    *
    * Please note that email is case insensitive, result for searching 'mail@email.com' or 'Mail@Email.com' will be the same
    */
-  public boolean doesEmailExist(DbSession dbSession, String email){
+  public boolean doesEmailExist(DbSession dbSession, String email) {
     return mapper(dbSession).countByEmail(email.toLowerCase()) > 0;
   }
 
   protected UserMapper mapper(DbSession session) {
     return session.getMapper(UserMapper.class);
   }
+
+  private static class LoginToUser implements Function<String, UserDto> {
+    private final Map<String, UserDto> map = new HashMap<>();
+
+    private LoginToUser(Collection<UserDto> unordered) {
+      for (UserDto dto : unordered) {
+        map.put(dto.getLogin(), dto);
+      }
+    }
+
+    @Nullable
+    @Override
+    public UserDto apply(@Nonnull String login) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)